File: publisher-service/src/main/java/com/souqstation/publisher/messaging/PublisherEventProducer.java
package com.souqstation.publisher.messaging;

import org.springframework.beans.factory.annotation.Value;
import org.springframework.kafka.core.KafkaTemplate;
import org.springframework.stereotype.Component;
import com.souqstation.schemas.events.GamePublished;
import org.apache.avro.generic.GenericRecord;

@Component
public class PublisherEventProducer {

    private final KafkaTemplate<String, Object> kafkaTemplate;
    private final String topic;

    public PublisherEventProducer(
            KafkaTemplate<String, Object> kafkaTemplate,
            @Value("${souq.topics.publisher}") String topic
    ) {
        this.kafkaTemplate = kafkaTemplate;
        this.topic = topic;
    }

    public void publishGame(String key, GamePublished event) {
        kafkaTemplate.send(topic, key, event);
    }

    public void publishPatch(String key, GenericRecord event) {
        kafkaTemplate.send(topic, key, event);
    }

    public void publishDlc(String key, GenericRecord event) {
        kafkaTemplate.send(topic, key, event);
    }
}

File: publisher-service/src/main/java/com/souqstation/publisher/messaging/PublisherEventProducer.kt
package com.souqstation.publisher.messaging

import com.souqstation.schemas.events.GamePublished
import com.souqstation.schemas.events.DLCPublishedEvent
import org.apache.avro.generic.GenericRecord
import org.springframework.beans.factory.annotation.Value
import org.springframework.kafka.core.KafkaTemplate
import org.springframework.stereotype.Component

@Component
class PublisherEventProducer(
    private val kafkaTemplate: KafkaTemplate<String, Any>,
    @Value("\${souq.topics.publisher}")
    private val topic: String
) {

    fun publishGame(key: String, event: GamePublished) {
        kafkaTemplate.send(topic, key, event)
    }

    fun publishPatch(key: String, event: GenericRecord) {
        kafkaTemplate.send(topic, key, event)
    }

    fun publishDlc(key: String, event: DLCPublishedEvent) {
        kafkaTemplate.send(topic, key, event)
    }
}